src/box/box.cc

```diff
@@ -4955,7 +5000,9 @@ void
 box_process_vote(struct ballot *ballot)
 {
 	ballot->is_ro_cfg = cfg_geti("read_only") != 0;
-	enum election_mode mode = box_check_election_mode();
+	const char *mode_name = cfg_gets("election_mode");
+	enum election_mode mode = election_mode_by_name(mode_name);
+	assert(mode != ELECTION_MODE_INVALID);
```

Comment:
Let's also add the following assertion: `assert(!(ballot->is_anon && ballot->can_lead));`. anon replica should never lead

Reply:
Added.